include/plugin-apis/impl-in-core/logging-abi.h

```diff
@@ -0,0 +1,193 @@
+// NON-ABI ************************************************************
```

Comment:
> it probably should be in a different header

You're right, the fact that there are two sections here indicates that there should be two different files. I'll take care of that. The one that contains the macros can be suffixed by `_macros` or `_api`, as you suggested.

> If it is not part of the API...

It seems like you already understand, but I'll add clarification here just to be extra sure.

It is kind of part of an API, but it doesn't work at the binary level. Maybe this distinction is clearer in the context of the discussion above.

This is messy. For the reasons described above, I want these headers to work as **much** at the binary level as possible and as **little** at the C syntax level as possible. But the fact is that these syntactic preprocessor things are useful, and so if another module does happen to be written in C, I want to have this stuff exposed.

Reply:
I don't think avoiding C syntax is a good goal. The goal should be to identify the actual (minimal) interface module and separate it as much as possible from convenience features on top.

Reply:
Marking as resolved since at this point we do indeed have two separate header files, `logging.h` and `logging_macros.h`, and because at this point we do seem to have a good shared understanding about what I mean about not working at the C syntax level and why.